src/metadataGeneration/tsoa.ts

```diff
@@ -68,8 +68,18 @@ export namespace Tsoa {
     validators: Validators;
   }
 
+  export type TypeStringLiteral = 'string' | 'boolean' | 'double' | 'float' | 'integer' | 'long' | 'enum' | 'array' | 'datetime' | 'date' | 'binary' | 'buffer' | 'byte' | 'void' | 'object' | 'any' | 'refEnum' | 'refObject';
+
+  export type RefTypeLiteral = 'refObject' | 'refEnum';
+
+  export type PrimitiveTypeLiteral = Exclude<TypeStringLiteral, RefTypeLiteral | 'enum' | 'array' | 'void'>;
+
   export interface Type {
-    dataType: 'string' | 'double' | 'float' | 'integer' | 'long' | 'enum' | 'array' | 'datetime' | 'date' | 'buffer' | 'void' | 'object' | 'any' | 'refEnum' | 'refObject';
+    dataType: TypeStringLiteral;
```

Comment:
I'm exporting the union of types so that I could do exhuastiveness checking. See the `assertNever` method I'm using https://github.com/lukeautry/tsoa/pull/385/files#diff-817ee96768e8e5542508a612054dfd39R1